k8s/k8s.go

```diff
@@ -321,25 +320,25 @@ func CmdDelK8s(c *calicoclient.Client, ep api.WorkloadEndpointMetadata, args *sk
 		// Check if ActiveInstanceID is populated (it will be an empty string "" if it was populated
 		// before this field was added to the API), and if it is there then compare it with ContainerID
 		// passed by the orchestrator to make sure they are the same, return without deleting if they aren't.
-	} else if wep.Metadata.ActiveInstanceID != "" && args.ContainerID != wep.Metadata.ActiveInstanceID {
+	} else if wep.Spec.ContainerID != "" && args.ContainerID != wep.Spec.ContainerID {
 		logger.WithField("WorkloadEndpoint", wep).Warning("CNI_ContainerID does not match WorkloadEndpoint ActiveInstanceID so ignoring the DELETE cmd.")
```

Comment:
nit (and not code you touched in this PR) but `CNI_ContainerID` is pretty bad - should be `ContainerID` or `CNI_CONTAINERID`